hibernate-core/src/main/java/org/hibernate/event/internal/EntityState.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.event.internal;
+
+import org.hibernate.engine.internal.ForeignKeys;
+import org.hibernate.engine.spi.EntityEntry;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.engine.spi.Status;
+import org.hibernate.internal.CoreLogging;
+import org.hibernate.internal.CoreMessageLogger;
+
+public enum EntityState {
+	PERSISTENT, TRANSIENT, DETACHED, DELETED;
+
+	static final CoreMessageLogger LOG = CoreLogging.messageLogger( EntityState.class );
+
+	/**
+	 * Determine whether the entity is persistent, detached, or transient
```

Comment:
`Determines` would be more consistent with existing Javadoc pattern, though not a big deal for sure.